### PR TITLE
Fix #APPINT-146 #APPINT-186 Add Payment + Opportunity

### PR DESCRIPTION
**Pull request by x4d3**

### CODE REVIEW

File: app/models/entities/sub_entities/payment.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  def self.object_name_from_connec_entity_hash(entity)
+    entity['title']
+  end
+
+  # Temp
+  def self.can_read_connec?
+    false
+  end
+
+  # Naming issue, can't have two sub entities called payment.rb
```
Comment (Berardpi) [inline app/models/entities/sub_entities/payment.rb]: Do you really need those?

File: app/models/entities/sub_entities/transaction.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    }
+  end
+
+  def self.object_name_from_external_entity_hash(entity)
+    entity['order_id']
+  end
+
+
+  def get_external_entities(last_synchronization = nil)
+    orders = @external_client.find('Order')
```
Comment (Berardpi) [inline app/models/entities/sub_entities/transaction.rb]: So you'll fetch the Orders twice at each sync?

File: app/models/maestrano/connector/rails/entity.rb [hunk 3]
```diff
@@ -24,6 +24,11 @@ def self.id_from_external_entity_hash(entity)
     entity['id']
   end
 
+  def self.creation_date_from_external_entity_hash(entity)
+    entity['created_at'] ? entity['created_at'].to_time : nil
```
Comment (Berardpi) [inline app/models/maestrano/connector/rails/entity.rb]: Do you have entities without timestamps?

File: spec/models/entities/financial_spec.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  describe 'instance methods' do
+    let(:organization) { create(:organization) }
+    let(:connec_client) { Maestrano::Connec::Client[organization.tenant].new(organization.uid) }
+    let(:external_client) { Maestrano::Connector::Rails::External.get_client(organization) }
+    let(:opts) { {} }
+    subject { Entities::Financial.new(organization, connec_client, external_client, opts) }
+
+
+    describe 'external_model_to_connec_model' do
+      expected =     {
```
Comment (Berardpi) [inline spec/models/entities/financial_spec.rb]: let(:expected) {}

File: spec/models/entities/financial_spec.rb [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+          'Order' => {'Sales Order' => [{}]},
+          'Shopify Invoice' => {'Invoice' => []},
+          'Transaction' => {
+              'Payment' => [],
+              'Opportunity' => []
+          }
+      }
+
+      it { expect(subject.external_model_to_connec_model({'Order' => [{}]})).to eql(expected) }
```
Comment (Berardpi) [inline spec/models/entities/financial_spec.rb]: Can you add a case with transactions?

File: app/models/maestrano/connector/rails/entity.rb (hunk 3 above)
Comment (x4d3) [inline app/models/maestrano/connector/rails/entity.rb]: No, I'll remove it.

**Pull request merged**